modules/hal_sensors/sensor_differentialpressure/src/sensor_differentialpressure.cpp

```diff
@@ -52,11 +53,35 @@ void DifferentialPressureSensor::BeginSDP810() {
 }
 
 void DifferentialPressureSensor::ReadSdp810() {
-  i2c_handle_->ReadBytes(sensor_buffer_, kSdp810BufferSize);
+  #define pi 3.141592653
+  #define m3tocm3_factor 1000000
+  #define air_weight 1.29
+  sampleTime = 10;
+  air_drag = 0.61;
+  hose_orifice_ratio = 0.5;
+  orifice_diameter = 0.003;
+  correction_factor = 1.525;
+  orifice_surface = pi / 4.0 * pow(orifice_diameter, 2);
 
+  // read and correct sensor
+  i2c_handle_->ReadBytes(sensor_buffer_, kSdp810BufferSize);
   conversion_factor_ = (sensor_buffer_[6] << (kSdp810BufferSize - 1) | sensor_buffer_[7]);
   sensor_raw_ = (sensor_buffer_[0] << (kSdp810BufferSize - 1) | sensor_buffer_[1]);
   sensor_raw_ = (sensor_raw_ / conversion_factor_);
+
+  if (sensor_raw_> 0) {
+  volume = (air_drag / sqrt(1-(hose_orifice_ratio * hose_orifice_ratio * hose_orifice_ratio * hose_orifice_ratio)) * orifice_surface * sqrt(2 * sensor_raw_ * air_weight)) * correction_factor; // measure amount of air blown into the lungs in this sample in ml
+  volume *= m3tocm3_factor;
+  volume /= 1000 / sampleTime;
+  volume_ += volume;
```

Comment:
It seems that volume_ is the accumulated value of volume over time, maybe give this variable another name for readability sake